lib/resolvers.js

```diff
@@ -0,0 +1,97 @@
+class Resolvers {
+  constructor(context) {
+    this.Kuzzle = context.accessors.sdk;
+
+    this._resolversMap = {
+      Query: {
+        allBooks: async () => {
+          const results = await this.Kuzzle.document.search(
+            'test-data',
+            'books'
+          );
+
+          if (!results.hits.length) {
+            throw new Error('No books were found!');
+          }
+          return results.hits.map((hit) => {
+            return { id: hit._id, ...hit._source };
+          });
+        },
+        book: async (root, { id }) => {
+          const result = await this.Kuzzle.document.get(
+            'test-data',
+            'books',
+            id
+          );
+
+          return { id: result._id, ...result._source };
+        },
+        author: async (root, { name }) => {
+          const authorResults = await this.Kuzzle.document.search(
+            'test-data',
+            'authors',
+            {
+              query: {
+                match: {
```

Comment:
The match query will fuzzy match so it can lead to erroneous results in this case. You should rather use a `term` query